src/use_case/collection/delete_item/DeleteItemInteractor.java

```diff
@@ -0,0 +1,48 @@
+package use_case.collection.delete_item;
+
+import entity.Ingredient;
+import entity.IngredientFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DeleteItemInteractor implements DeleteItemInputBoundary{
+
+    final DeleteItemDataAccessInterface deleteItemDataAccessObject;
+
+    final DeleteItemOutputBoundary deletePresenter;
+
+    final IngredientFactory ingredientFactory;
+
+    public DeleteItemInteractor(DeleteItemDataAccessInterface deleteItemDataAccessObject,
+                                DeleteItemOutputBoundary deletePresenter,
+                                IngredientFactory ingredientFactory) {
+        this.deleteItemDataAccessObject = deleteItemDataAccessObject;
+        this.deletePresenter = deletePresenter;
+        this.ingredientFactory = ingredientFactory;
+    }
+
+//know when to have success view and fail view.
```

Comment:
add more detail about the condition for success and fail view

Reply:
done